### PR TITLE
Throw errors and fix spelling in math

### PR DESCRIPTION
**Pull request by iwei20**

### CODE REVIEW

File: src/com/stuypulse/stuylib/math/Vector2D.java [hunk 1]
```diff
@@ -24,7 +24,10 @@ public Vector2D() {
      * Initialize Vector2D using an array of 2 doubles
      * @param in an array of 2 numbers
      */
-    public Vector2D(double[] in) {
+    public Vector2D(double[] in) throws Exception {
```
Comment (CreativePenguin) [inline src/com/stuypulse/stuylib/math/Vector2D.java]: Don't use a checked exception here. Use RuntimeException instead. I personally hate checked exceptions, but this isn't the kind of error that the user will know how to fix.

Comment (CreativePenguin) [reply src/com/stuypulse/stuylib/math/Vector2D.java]: I will say, I don't think this constructor needs to exist.

File: src/com/stuypulse/stuylib/math/streams/filters/MovingAverage.java [hunk 2]
```diff
@@ -19,10 +19,14 @@
 
     /**
      * Make Simple Moving Average with Max Array Size
-     * @param size size of moving average
+     * @param size size of queue of values calculated for moving average
      */
-    public MovingAverage(int size) {
-        mSize = Math.max(size, 1);
+    public MovingAverage(int size) throws Exception {
```
Comment (CreativePenguin) [inline src/com/stuypulse/stuylib/math/streams/filters/MovingAverage.java]: Also change this into runtime exception

File: src/com/stuypulse/stuylib/math/Vector2D.java (hunk 1 above)
Comment (Sam-Belliveau) [reply src/com/stuypulse/stuylib/math/Vector2D.java]: this constructor is based off of the way some api's represent vector 2d

**Pull request merged**